remote-api/src/main/java/org/alfresco/repo/web/scripts/rule/AbstractRuleWebScript.java

```diff
@@ -432,10 +442,40 @@ private Serializable convertValue(QName typeQName, Object propertyValue) throws
 
     protected void checkRule(Rule rule)
     {
+        List<Action> actions = getAllActions(rule);
+        blockPrivateActions(actions);
+        validateOutboundRuleActions(rule, actions);
+    }
+
+    private List<Action> getAllActions(Rule rule) {
+        List<Action> actionList = new ArrayList<>();
+        Action action = rule.getAction();
+
+        if (action instanceof CompositeAction) {
+            actionList.addAll(((CompositeAction) action).getActions());
+        } else {
+            actionList.add(action);
+        }
+
+        return actionList;
+    }
+
+    private void blockPrivateActions(List<Action> actions) {
+        for (Action action : actions)
+        {
+            ActionExecutionContext aec = ActionExecutionContext.builder(action.getActionDefinitionName())

```

Comment:
Was the intent behind the check about "action.getActionDefinitionName()" as I've assumed? For action.getId() all tests failed and it also seemed counter-intuitive

Reply:
Yes, the `getActionDefinitionName` returns the bean id which is used to identify the logic behind the action.